R/plotPrevalence.R

```diff
@@ -235,14 +253,31 @@ setMethod("plotPrevalentAbundance", signature = c(x = "SummarizedExperiment"),
         xlab <- paste0(ifelse(as_relative, "Rel. ", ""),"Abundance")
         ylab <- paste0("Prevalence(", ifelse(is.null(rank), "Features", rank),
                        ") [%]")
-        .prevalence_plotter(plot_data, 
+        plot <- .prevalence_plotter(plot_data, 
                             layout = "point",
                             xlab = xlab,
                             ylab = ylab,
                             colour_by = colour_by,
                             size_by = size_by,
                             shape_by = shape_by,
                             ...)
+        
+        
+        # If facet_by is not FALSE, user has specified it. Adds the facets to the plot.
+        if(facet_by != FALSE){
+            plot <- plot + 
+                # Changes the theme so the plot is easier to read. Adds e.g. outlines
+                theme_bw() +
+                theme(plot.background=element_blank(), 
+                      panel.grid.major=element_blank(),
+                      panel.grid.minor=element_blank(), 
+                      panel.background=element_blank(),
+                ) +
+                # Create facets
+                facet_wrap(~as.factor(rowData(x)[[facet_by]])) 
+        }
+        
+        return(plot)
```

Comment:
I think you should integrate this into `.incorporate_prevalence_vis` instead of bolting it on like this. If you choose to keep it like this, you should make sure, that the contract, that all data relevant for plotting is present in `object`, applies to this case as well.

Reply:
Yep, that is better to have all the data in the object

I added facet_by also to .incorporate_prevalence_vis